Assignment 1B/script.js

```diff
@@ -0,0 +1,118 @@
+    let ageEntry = ""; //declaring and initializing variable to store age selected
+
+    function catchFirst(event){ //method to store user entry of first name
+        firstText = event.target.value; //storing user entry in firstText
+    }
+
+    function catchLast(event){ //method to store user entry of last name
+        lastText = event.target.value; //storing user entry in lastText
+    }
+
+    function catchAge(event){ //method to store user selection of age
+        ageEntry = event.target.value; //storing user entry in ageEntry
+    }
+
+    addEventListener("submit", submitFunction); //eventListener added on submit -- calls submitFunction method
+
+    function submitFunction(){ //submitfunction is the primary method that controls the form validation
+        preventSubmitChecker(event); //method called to check if data entered is invalid and if so prevents form from submitting
+        checkDiv(errorDiv); //method called to check if the errorDiv already contains elements -- removes them if it does
+        createErrorDisplay(); //method called to check which entries are invalid and create the errorDiv display
+    }
+
+    function checkEntry(userEntry){ //checkEntry method accepts user entry parameter and checks if it contains any string value
+        let formValue = userEntry; //storing the user entry in formValue variable 
+        if(formValue === ""){ //if statement to determine return value
+            return false; //returns false if the user entry did not contain characters
+        } else { //else statement to determine alternate return value
+            return true; // returns true if there is indeed a character
+        }
+    }
+
+    function preventSubmitChecker(event){ //function accepts event as parameter -- this function checks whether data is valid and if not it prevents the form submission
+        let a = checkEntry(firstText); //storing first name data validity return value into variable
+        let b = checkEntry(lastText); //storing last name data validity return value into variable
+        let c = checkEntry(ageEntry); //storing age data validity return value into variable
+
+        if (a === false || b === false || c === false) { //checks if any of the user entry values are invalid
+            event.preventDefault(); //if user entry is invalid, uses preventDefault method to stop submission
+        }
+    }
+
+    function createErrorDisplay(){ //this function creates the error messages
+        let a = checkEntry(firstText); //checks data validity
+        let b = checkEntry(lastText); //checks data validity
+        let c = checkEntry(ageEntry); //checks data validity
+        let errorList = document.createElement("ul"); //created ul element to be appended in first if statement --  it is declared here to allow all parts of this function access
+
+        if(a === false || b === false || c === false){ //if statement checking returned variables from above to see if any data was declared invalid 
```

Comment:
you preform this same if eval in preventSubmitChecker() - this is another of those opportunities for efficiency if you combined the functions.